app/models/notification.rb

```diff
@@ -96,13 +96,13 @@ def send_milestone_notification(type:, article_id:)
     def remove_all_by_action(notifiable_ids:, notifiable_type:, action: nil)
```

Comment:
I don't think we are calling this method anywhere. Can you double-check? If indeed we are not we should probably remove it. 

If we remove this method then we don't ever need notifications updated in the background so we may as well remove the worker and call the service inline. What do you think @rhymes? I would rather wait until we need something at this point than add complexity we are not using.

Reply:
Awesome catch @mstruve! I double checked:

➜  devto git:(master) rg "\bremove_all_by_action\b"
app/models/notification.rb
96:    def remove_all_by_action(notifiable_ids:, notifiable_type:, action: nil)

which means that we can remove the method:

https://github.com/thepracticaldev/dev.to/blob/6321fbac9c3abd2600c49f9050ebd710b3ba1912/app/models/notification.rb#L96-L100

which means that in this method, instea of calling the worker like this:

https://github.com/thepracticaldev/dev.to/blob/6321fbac9c3abd2600c49f9050ebd710b3ba1912/app/models/notification.rb#L102-L106

we can just:

```ruby
def remove_all_by_action_without_delay(notifiable_ids:, notifiable_type:, action: nil)
  return unless %w[Article Comment Mention].include?(notifiable_type) && notifiable_ids.present?

  Notifications::RemoveAllByAction.call(Array.wrap(notifiable_ids), notifiable_type, action)
end
```

and thus delete the job and its spec:

- https://github.com/thepracticaldev/dev.to/blob/master/app/jobs/notifications/remove_all_by_action_job.rb
- https://github.com/thepracticaldev/dev.to/blob/master/spec/services/notifications/remove_all_by_action_spec.rb

sorry @luchiago, I totally didn't catch this and asked you to implement more specs :(

how would you feel at implementing this change?

Reply:
@rhymes That's no problem. I feel happy to help you guys to detect this and all knowledge about sidekiq is worth it. Should I send another PR deleting the old job and spec?

Reply:
@rhymes No problem for me. I feel happy helping you guys to detect this and fix it. Should I delete the old job and spec in this PR or in another?

Reply:
And the worker won't be necessary anymore?

Reply:
Correct, the worker won't be necessary if you directly call the action.

As `Notification.remove_all_by_action_without_delay` is only called when an article is unpublished you can also remove the old job and its spec.

Let's clean them all up!

Thank you @luchiago

Reply:
You're welcome @rhymes. I'll send the changes.